### ISSUE TITLE
Some queries never finish although they stopped to generate data

### ISSUE THREAD

**Issue by josephnoir** [vast-io/vast#130]
Queries on "large" databases seem return data but never finish. I am playing around a bit with data collected with Bro and this is the related query `vast export bro -h "(id.resp_p == 137/? || id.resp_p == 138/? || id.resp_p == 445/?) && proto == \"tcp\""`. Here is the importer log:
```
Executing first query, looking for connection attempts on ports (137, 138, 445) using TCP
     _   _____   __________
    | | / / _ | / __/_  __/
    | |/ / __ |_\ \  / /
    |___/_/ |_/___/ /_/  0.1

connecting to 127.0.0.1:42000
spawning sink with parameters: ("bro", "-h", "(id.resp_p == 137/? || id.resp_p == 138/? || id.resp_p == 445/?) && proto == \"tcp\"")
spawning exporter with parameters: ("exporter", "-h", "(id.resp_p == 137/? || id.resp_p == 138/? || id.resp_p == 445/?) && proto == \"tcp\"")
signal-monitor sends signals to 8
signal-monitor registers signal handler for Hangup
signal-monitor registers signal handler for Interrupt
signal-monitor registers signal handler for Quit
signal-monitor registers signal handler for Terminated
signal-monitor registers signal handler for User defined signal 1
signal-monitor registers signal handler for User defined signal 2
bro-writer got query statistics from 6eb11d79-5ada-4ff4-9d44-6e545499ad21
bro-writer got query statistics from 6eb11d79-5ada-4ff4-9d44-6e545499ad21
bro-writer got query statistics from 6eb11d79-5ada-4ff4-9d44-6e545499ad21
bro-writer creates a new stream for STDOUT
```

Due to its length, the vast log is in this [gist](https://gist.github.com/josephnoir/53bd7f876fca239725ec8118d717f744) and not complete. My terminal scrollback is limited. Think I mentioned the behavior in the channel, but failed to create an Issue for it.

**Comment by MarcinNaw** [comment]
If you need additional data for debugging this issue (which I also experience quite frequently), I could provide data.

While working on the same VAST state, queries for some ports terminate, however for port 80 they never do.

**Comment by mavam** [comment]
Data for reproducing would be really helpful. I could not reliably trigger this by trial and error.

**Comment by josephnoir** [comment]
Maybe I can provide some, see my comment in Gitter. Although more data may always be more helpful ^^

**Comment by mavam** [comment]
Thanks @josephnoir, that was helpful. (For the record, this was a private Gitter discussion.)

**Comment by MarcinNaw** [comment]
I have also provided a data set in a private Gitter discussion, good luck on debugging!

**Comment by mavam** [comment]
Sorry that this one was on the backburner for so long. Would you mind sending me the test data to reproduce the bug again?

**Comment by josephnoir** [comment]
I posted a log to the VAST Gitter chat, hope it's still there to download.

**Comment by mavam** [comment]
Thanks. @MarcinNaw also provided me with another sample.

**Comment by Neverlord** [comment]
Turned out to be a bug in CAF. Fixed in `master`.

**Comment by Neverlord** [comment]
Sorry, the CAF bug was about VAST not terminating properly.

**Comment by mavam** [comment]
We've narrowed down the bug origin: the problem is that the archive returns less events than requested: the index lookup for `id.resp_p == 80/?` returns 35,462 hits, but the archive then only returns 35461! The missing event has ID 196,608. Given that the archive contains a total of 270,930 events, this means that the missing event is somewhere in the middle. But the archive contains all events, without a gap. So this is a bug in the archive. More soon.

**Comment by mavam** [comment]
Fixed with 6fa16474fd00938fd26d0c0d2906b2a17663dc62.
